compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -2748,6 +2761,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 _ => {}
             }
         }
+
+        // We failed to check the expression, report an error.
```

Comment:
This comment doesn't really describe what this is doing. It's attempting to emit an error if we end up derefing to an infer var, like calling `.field` on a base type of `&_`.

Reply:
That comment was more general, as in "ok we're no longer in the happy path"

Reply:
Well it's useful regardless to document the random structural resolve call. Pls do that :)